### ISSUE TITLE
Support option to sort alphabetically without taking capitalization in to consideration

### ISSUE THREAD

**Issue by fiveNinePlusR** [Alexander-Miller/treemacs#158]
It would be nice to have an option that downcases every file before deciding on sort order. This would fix the problem where capitalized folders appear higher in the list than lower case folders/files... effectively splitting each folder into four groups of display: 

1. capitalized folders 
2. lowercase folders
2. capitalized files
2. lower case files

which is confusing in general. I know we talked about the performance repercussion of this so it would need to be an alternative sort option and not set as the default.  I will see if I can get a patch for this feature soon.

**Comment by Alexander-Miller** [comment]
Adding that option should be easy enough. Your entry point is [over here](https://github.com/Alexander-Miller/treemacs/blob/master/src/elisp/treemacs-customization.el#L189). You'd just need to add `alphabetic-asc/desc-ignore-case` options.
If you need help getting around the code base I am happy to help.

**Comment by Alexander-Miller** [comment]
To answer your questions from gitter:

 * Don't know about pfuture. I just tried cask install after deleting .cask and run into `Variable binding depth exceeds max-specpdl-size`, which seems to be an issue with shut-up.el. It did work after trying a second time. So try again I guess.

 * No tests for sorting. I'll put it on my todo list to add some to the systest, but that's only after multiroot is done, since the sys test will need to be largely rewritten anyway.

**Comment by fiveNinePlusR** [comment]
I did get cask working after some fiddling about. all seems good now.
